test/metabase/formatter_test.clj

```diff
@@ -11,14 +11,26 @@
                                      {{::mb.viz/field-id 1} viz}})
         value)))
 
+(defn format-with-merged-settings
+  "Include both field-id and column-name keys in column settings #55066"
+  [value viz]
+  (str ((formatter/number-formatter {:id 1 :name "FOO"}
+                                    {::mb.viz/column-settings
+                                     {{::mb.viz/field-id 1} {::mb.viz/view-as nil}
+                                      {::mb.viz/column-name "FOO"} viz}})
+        value)))
+
 (deftest number-formatting-test
   (let [value 12345.5432
-        fmt   (partial format value)]
+        fmt   (partial format value)
+        fmt2  (partial format-with-merged-settings value)]
```

Comment:
This seems like a good place to use `doseq` to run each case below with all three permutations (just id, just name, and both).